### DISCUSSION TITLE
Get all variation of a product in one API

### DISCUSSION THREAD

**Discussion by kishanK111** [amzn/selling-partner-api-models#3490] [General]
I am working with catalogue listing API by searching keywords. But when i got product as a parent product and it has many child ASINs.
I have to call API for every child ASINs. It is consuming so many time. Is there any API to get all child product from giving ASIN of parent product?

`
```
{
       "asin": "B0BRQ1T9GT",
       "relationships": [
                    {
                        "marketplaceId": "A21TJRUUN4KGV",
                        "relationships": [
                            {
                                "childAsins": [
                                    "B0BQ3RC28F",
                                    "B0BQ3MMPX6",
                                    "B0BQ3PJJDQ",
                                    "B0BQ3Q5NDH",
                                    "B0BQ3MT8P2",
                                    "B0BQ3PYMCZ"
                                ],
                                "type": "VARIATION",
                                "variationTheme": {
                                    "attributes": [
                                        "color",
                                        "style"
                                    ],
                                    "theme": "STYLE_NAME/COLOR_NAME"
                                }
                            }
                        ]
                    }
                ]
}

````

**Comment by kimeseonhoo1** [comment]
Hi @Kishank111  , I have the same problem, did you ever find solution how to get product with all variations, or maybe all listings of one merchant.

**Comment by stefnats** [comment]
I don't have a solution, but you could batch up to 20 ASINs with the [`searchCatalogItems`](https://developer-docs.amazon.com/sp-api/docs/catalog-items-api-v2022-04-01-reference#searchcatalogitems) call instead of the `getCatalogItem` call (only one at a time).

**Reply by kimeseonhoo1** [reply]
Hi, but I need to get all product that one merchant has published

**Reply by stefnats** [reply]
Oh i misunderstood your question then. 

Then your best chance is to request the report type `GET_MERCHANT_LISTINGS_ALL_DATA` 
If you're still missing variation related information, you have to request via Catalog Items API. Listings Reports are limited in catalog (asin related) data.

**Comment by jasonhebert** [comment]
Not sure if this will give you anything that you would need but to get all listings for a seller you can use the Report API and request report type `GET_MERCHANT_LISTINGS_ALL_DATA`. This will give you a tab-delimited flat file detailed all listings report.

**Comment by kevnhowe** [comment]
Seems like you would query your parent for the child ASINs, then iterate over your child ASINs with a getcatalogitem request to return the specific attribute values associated with the variation relationship for each child. However, I have found no way to return attribute values of any kind for my queries. I asked last month: https://github.com/amzn/selling-partner-api-docs/discussions/3517#discussion-6014267 and received no response. Not sure if the API is missing this functionality, or if I'm missing something in my request.
